### PR TITLE
Profile pairing (depends on #59)

### PR DESCRIPTION
**Pull request by dtebbs**
Add profiling code for pairing computations.

### CODE REVIEW

File: libff/algebra/curves/profile/algebra_groups_profile.cpp [hunk 1]
```diff
@@ -111,6 +111,79 @@ template<typename GroupT> bool profile_group_membership()
...
+        const G1_precomp<ppT> sig_prec = ppT::precompute_G1(signatures[i]);
+        const G1_precomp<ppT> H_precomp =
+            ppT::precompute_G1(messages[i] * G1<ppT>::one());
+
+        const Fqk<ppT> result =
+            ppT::final_exponentiation(ppT::double_miller_loop(
+                sig_prec, id_precomp, H_precomp, pk_precomp));
+
+        if (i & 1) {
+            if (Fqk<ppT>::one() != result) {
```
Comment (AntoineRondelet) [inline libff/algebra/curves/profile/algebra_groups_profile.cpp:167]: `if (Fqk<ppT>::one() != result) {` -> `if (result != Fqk<ppT>::one()) {` for consistency

File: libff/algebra/curves/profile/algebra_groups_profile.cpp [hunk 2]
```diff
@@ -111,6 +111,79 @@ template<typename GroupT> bool profile_group_membership()
...
+                throw std::runtime_error(
+                    "signature verification failed " + std::to_string(i));
+            }
+        } else {
+            if (Fqk<ppT>::one() == result) {
```
Comment (AntoineRondelet) [inline libff/algebra/curves/profile/algebra_groups_profile.cpp:172]: `if (Fqk<ppT>::one() == result) {` -> `if (result == Fqk<ppT>::one()) {`

**Comment by dtebbs** [comment]
@AntoineRondelet thanks for the comments.  Should be addressed in the latest version.

**Review by AntoineRondelet — APPROVED** [review]
LGTM, thanks @dtebbs

**Pull request merged**